google_guest_agent/network/managers.go

```diff
@@ -0,0 +1,85 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/GoogleCloudPlatform/guest-agent/google_guest_agent/run"
+	"github.com/GoogleCloudPlatform/guest-agent/metadata"
+)
+
+// ManagerService is an interface for setting up network configurations
+// using different network managing services, such as systemd-networkd and wicked.
+type ManagerService interface {
+	IsManaging(ctx context.Context, iface string) (bool, error)
+	Setup(ctx context.Context, payload []metadata.NetworkInterfaces) error
+	Rollback(ctx context.Context, payload []metadata.NetworkInterfaces) error
+}
+
+var (
+	badMAC               = []string{}
+	googleComment        = "# Added by Google Compute Engine OS Login."
```

Comment:
Should this be `Guest Agent` instead of `OS Login`?

Reply:
Done.